vispy/scene/cameras/cameras.py

```diff
@@ -71,43 +197,191 @@ def interactive(self):
         
-    def view_changed(self):
-        """ Called when this camera is connected to a new view.
+        E.g. (0.5, 1.0, 1.0) would make the x-dimension half as small.
+        Negative values can be used to flip dimensions, e.g. (1, -1)
+        is typically suited for visualizing images.
+        """
+        return self._aspect_ratio
+    
+    @aspect_ratio.setter
+    def aspect_ratio(self, value):
+        # Check and prepare
+        if len(value) == 2:
+            value = float(value[0]), float(value[1]), 1.0
+        elif len(value) == 3:
+            value = float(value[0]), float(value[1]), float(value[2])
+        else:
+            raise ValueError('Aspect ratio must be a 2 or 3 element tuple')
+        # Normalize
+        self._aspect_ratio = value
+        self._aspect_ratio_n = tuple([v / abs(value[0]) for v in value])
+        self.view_changed()
+    
+    @property
+    def aspect_fixed(self):
+        """ Whether the aspect ratio is fixed
+        
+        If not fixed, the camera is allowed to modify the aspect ratio
+        during resetting, window resizing, and interaction.
+        """
+        return self._aspect_fixed
+    
+    @aspect_fixed.setter
+    def aspect_fixed(self, value):
+        self._aspect_fixed = bool(value)
+    
+    @property
+    def center(self):
+        """ The center location for this camera
+        
+        The exact meaning of this value differs per type of camera, but
+        generally means the point of interest or the rotation point.
+        """
+        return self._center_loc or (0, 0, 0)
+    
+    @center.setter
+    def center(self, val):
+        if len(val) == 2:
+            self._center_loc = float(val[0]), float(val[1]), 0.0
+        elif len(val) == 3:
+            self._center_loc = float(val[0]), float(val[1]), float(val[2])
+        else:
+            raise ValueError('Center must be a 2 or 3 element tuple')
+        self.view_changed()
+    
+    @property
+    def fov(self):
+        """ Field-of-view angle of the camera. If 0, the camera is in
+        orthographic mode.
+        """
+        return self._fov
+    
+    @fov.setter
+    def fov(self, fov):
+        fov = float(fov)
+        if fov < 0 or fov >= 180:
+            raise ValueError("fov must be between 0 and 180.")
+        self._fov = fov
+        self.view_changed()
+    
+    def set_range(self, xrange=None, yrange=None, zrange=None, margin=0.05):
+        """ Set the range of the view region for the camera
+        
+        The view is reset to the given range or to the scene boundaries
+        if ranges are not specified. The ranges should be 2-element
+        tuples specifying the min and max for each dimension.
+        """
+        # Do we have a viewbox?
+        if self._viewbox is None:
+            return
+        self._resetting = True
+        # Size factor
+        size = self._viewbox.size
+        self._window_size_factor = size[1] / size[0]
```

Comment:
I am not sure what you mean. `window_size_factor` (only used internally) applies to the window. `aspect_ratio` applies to the local coordinate frame. E.g. if the user resizes the window, the former is changed, but the latter should remain equal, and what was previously square should still be square.